src/app/pages/PerpetualPage/hooks/usePerpetual_openTrade.ts

```diff
@@ -25,14 +25,25 @@ const MASK_CLOSE_ONLY = 0x80000000;
 export const usePerpetual_openTrade = () => {
   const address = useAccount();
 
-  const { ammState, perpetualParameters } = useContext(PerpetualQueriesContext);
+  const {
+    ammState,
+    perpetualParameters,
+    depthMatrixEntries: entries,
+  } = useContext(PerpetualQueriesContext);
 
   const marginBalance = usePerpetual_marginAccountBalance();
   const midPrice = useMemo(() => getMidPrice(perpetualParameters, ammState), [
     perpetualParameters,
     ammState,
   ]);
 
+  let averagePrice = undefined;
+
+  if (entries && entries.length >= 3) {
+    const averagePriceIndex = entries[1].indexOf(0);
+    averagePrice = entries[0][averagePriceIndex];
+  }
```

Comment:
I really dislike this solution. We will also need to implement it in the trade form and close modal to show the correct limit values. Either we add a utils function or just add the extracted averagePrice to the context

Reply:
I decided to add it to the context and I also used it in the trade form and slippage modals.